src/mintpy/tropo_local_texture.py

```diff
@@ -0,0 +1,321 @@
+############################################################
+# Program is part of MintPy                                #
+# Copyright (c) 2013, Zhang Yunjun, Heresh Fattahi         #
+# Author: Yang Qingyue, Hu Changyang, 2024                 #
+############################################################
+
+
+import os
+
+import numpy as np
+import scipy
+from scipy.interpolate import UnivariateSpline, griddata
+
+from mintpy.objects import timeseries
+from mintpy.utils import readfile, writefile
+
+############################################################################
+
+def read_topographic_data(geom_file):
+    print('read height from file: '+geom_file)
+    dem = readfile.read(geom_file, datasetName='height', print_msg=False)[0]
+    Na, Nr = dem.shape
+    dataLine = dem.flatten()
+
+    index = np.where(~np.isnan(dataLine))[0]
+    x = np.ceil(index / Na).astype(int)
+    y = np.mod(index, Na).astype(int)
+
+    xi = np.arange(0, Nr)
+    yi = np.arange(0, Na)
+
+    dataInp = griddata((x, y), dataLine[index], (xi[None, :], yi[:, None]), method='cubic')
+    dem[np.isnan(dem)] = dataInp[np.isnan(dem)]
+    return dem
+
+def estimate_local_slope(dem, ts_data, inps, n_ref, meta):
```

Comment:
I think k_left = -rg * step + coe[0] ,coe is not defined in line 154 @Yohcy 
I think instead of mask ,you mean temporal coherence,No?

Reply:
> I think k_left = -rg * step + coe[0] ,coe is not defined in line 154 @Yohcy I think instead of mask ,you mean temporal coherence,No?

@mohseniaref  Thanks for your review! I think coe is defined in line 126. For the second problem, the original author Yang calls it coherence mask, but that's temporal coherence indeed. Maybe I can change it later.